### DISCUSSION TITLE
How to input images to meshroom_compute?

### DISCUSSION THREAD

**Discussion by juanjosse** [alicevision/Meshroom#1669] [Q&A]
I have a custom graph I would like to run through command line using "meshroom_compute". But I need to be able to input different images each time.
The grapfile.mg is a JSON which contains the images I loaded in the GUI, but for substituing them with the new, I don't know how to fill some parameters that meshroom fills automatically about the intrinsics and so on (focal length, principal point, sensor height and width,...). Don't know how meshroom calculates these parameters from the images as a init guess.

When I do it from the GUI or with "meshroom_batch" I can input a folder of images, how to do with "meshroom_compute"?

How "meshroom_compute" is supposed to run a graph each time with different images (without creating new graphfiles.mg through GUI)?

**Comment by natowi** [comment]
You could use `--paramOverrides NODE.param=value` and override the cameraInit input folder

https://meshroom-manual.readthedocs.io/en/latest/feature-documentation/cmd/photogrammetry.html

Similar issue: https://github.com/alicevision/meshroom/issues/914

**Reply by juanjosse** [reply]
But meshroom_compute does not have this argument, this is for meshroom_batch and I need meshroom_compute to compute a custom graphfile.mg

**Reply by juanjosse** [reply]
Ok meshroom_batch can have an input --pipeline graphfile.mg, but how can I input a list of images? The help says -i : file .sfm or .json with images paths and optionally intrinsics. **Which format is it supposed to have?**

--input [SFM/FOLDERS/IMAGES [SFM/FOLDERS/IMAGES ...]]
                        Input folder containing images or folders of images or
                        **file (.sfm or .json) with images paths and optionally
                        predefined camera intrinsics.**
